Sources/AWSAppSyncApolloExtensions/Websocket/AppSyncWebSocketClient.swift

```diff
@@ -79,36 +79,46 @@ public class AppSyncWebSocketClient: NSObject, ApolloWebSocket.WebSocketClient,
     }
 
     public func connect() {
-        AppSyncApolloLogger.debug("Calling Connect")
-        guard connection?.state != .running else {
-            AppSyncApolloLogger.debug("[AppSyncWebSocketClient] WebSocket is already in connecting state")
-            return
-        }
-
-        subscribeToAppSyncResponse()
-
-        Task {
+        taskQueue.async { [weak self] in
+            guard let self else { return }
+            AppSyncApolloLogger.debug("Calling Connect")
+            guard connection?.state != .running else {
+                AppSyncApolloLogger.debug("[AppSyncWebSocketClient] WebSocket is already in connecting state")
+                return
+            }
+            
+            subscribeToAppSyncResponse()
+            
             AppSyncApolloLogger.debug("[AppSyncWebSocketClient] Creating new connection and starting read")
-            self.connection = try await createWebSocketConnection()
+            self.connection = try await self.createWebSocketConnection()
+            
             // Perform reading from a WebSocket in a separate task recursively to avoid blocking the execution.
-            Task { await self.startReadMessage() }
+            Task {
+                await self.startReadMessage()
```

Comment:
Should we store this task somewhere and clear it out upon new connection? I think it may be possible to have multiple of these running

Reply:
The task will be started just once per instance of `AppSyncWebSocketClient`

```
taskQueue.async { }
```
serializes the call to `connect()`.
 
All later invokes will return from 
```
guard connection?.state != .running else {
    AppSyncApolloLogger.debug("[AppSyncWebSocketClient] WebSocket is already in connecting state")
    return
}
```